### PR TITLE
Action bar directives

### LINKED ISSUE #55: Allow for ActionBar usage
Currently using an ActionBar in a {N}/ng2 app is a bit clunky (see https://github.com/NativeScript/sample-Groceries/blob/5b09135256c24ec4e8ba17b3dc76eff5de7db6cc/app/views/login/login.ts#L17-L19).

Ideally we’d have some way you could provide the necessary ActionBar XML elements in your Angular components. I’m not sure how we could make that possible, but I thought I’d open things up for discussion because I think this is going to be a common pain point.

### PR DESCRIPTION
**Pull request by vakrilov**
Resolves #55 
Ping: @hdeshev @nsndeck

### CODE REVIEW

**Comment by NathanWalker** [comment]
Ooooohhh ma goodness. This is exciting. @vakrilov Awesome.

**Comment by matt4446** [comment]
@vakrilov Nice
... and I was taking a short break from coding in the evenings..

File: src/nativescript-angular/directives/action-bar.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+registerElement("ActionBar", () => require("ui/action-bar").ActionBar, actionBarMeta);
+registerElement("ActionItem", () => require("ui/action-bar").ActionItem);
+registerElement("NavigationButton", () => require("ui/action-bar").NavigationButton);
+
+@Component({
+    selector: "ActionBar",
+    template: "<ng-content></ng-content>"
+})
+export class ActionBarDirective {
```
Comment (hdeshev) [inline src/nativescript-angular/directives/action-bar.ts]: Rename to ActionBarComponent? Not a directive anymore.

**Comment by hdeshev** [comment]
Just had a chat with @vakrilov and we decided to rename `<action-bar-scope>` to `<ActionBarExtension>`

File: src/nativescript-angular/directives/action-bar.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    constructor(public element: ElementRef, private page: Page) {
+        console.log("ActionBarDirective.constructor: " + element.nativeElement)
```
Comment (hdeshev) [inline src/nativescript-angular/directives/action-bar.ts]: We should switch all those `console.log` statements to `util.traceLog`

**Comment by jdnichollsc** [comment]
Guys, what's the way to change the title using the *ActionBarExtension* component?

**Pull request merged**